tests/integration/test_tls.py

```diff
@@ -223,6 +223,19 @@ async def test_mtls(ops_test: OpsTest):
     assert max_offset == str(num_messages)
 
 
+@pytest.mark.abort_on_fail
+async def test_mtls_broken(ops_test: OpsTest):
+    remove_relation_cmd = f"remove-relation {CHARM_KEY} {TRUSTED_CERTIFICATE_RELATION}"
+    await ops_test.juju(*remove_relation_cmd.split(), check=True)
+
+    await ops_test.model.wait_for_idle(
+        apps=[CHARM_KEY],
+        status="active",
+        idle_period=30,
+        timeout=2000,
+    )
```

Comment:
**todo** since this was happening also for `trusted-ca`, could we do this for that endpoint as well? I understand that the implementation is the same, but the test should probably do that for both, as we might change the implementation but the UX should be covered. 

I believe it would eventually be nice to have a tests for a `trusted-ca` beside the one for `trusted-certificates`, but I would leave it out the scope of the PR. But just having adding relation, removing relation can be beneficial since the bug had also that use case

Reply:
I agree in theory, but in practice the implementation **is** the same for both.

Here, we're testing `trusted-relation-broken`, which applies to both relation interfaces, with exactly the same logic underpinning it:
```python
for relation in [TRUSTED_CERTIFICATE_RELATION, TRUSTED_CA_RELATION]:
            self.framework.observe(
                self.charm.on[relation].relation_broken,
                self._trusted_relation_broken,
            )
```

Given this, it's fine to keep out. If we ever have branching logic for the two, then we work on adding tests for that logic in those PRs.

Reply:
yes, I know, that's why I said 

>  I understand that the implementation is the same

but it is generally good (whenever possible) to keep tests independent from the implementation. Hence the comment. But it is not a strong point if you feel otherwise.